PyTDF/Operation.py

```diff
@@ -1,38 +1,45 @@
 from __future__ import print_function
+from enum import Enum
 
 class Operation(object):
-    
     """
     A Generic representation
     of an operation such as 
     transformations and actions
 
     """
 
+    TYPES = Enum("TYPES", "ACTION TRANSFORMATION")
+
     def __init__(self, name, *args, **kwargs):
         self.name = name
         self.args = args
         self.kwargs = kwargs
         self.op_type = self._classify_operation(name)
 
     def _classify_operation(self, name):
+        if name.lower() in ("range", "take"):
+            raise NotImplementedError("This operation ({}) isn't supported yet ! ".format(name))
+
         operations_dict = {
-        'Define':'t',
-        'Filter':'t',
-        'Range':'t',
-        'Histo1D':'a',
-        'Histo2D':'a',
-        'Histo3D':'a',
-        'Profile1D':'a',
-        'Profile2D':'a',
-        'Profile3D':'a',
-        'Count':'a',
-        'Min':'a',
-        'Max':'a',
-        'Mean':'a',
-        'Sum':'a',
-        'Fill':'a',
-        'Take':'a'
+        'Define':Operation.TYPES.TRANSFORMATION,
+        'Filter':Operation.TYPES.TRANSFORMATION,
+        'Histo1D':Operation.TYPES.ACTION,
+        'Histo2D':Operation.TYPES.ACTION,
+        'Histo3D':Operation.TYPES.ACTION,
+        'Profile1D':Operation.TYPES.ACTION,
+        'Profile2D':Operation.TYPES.ACTION,
+        'Profile3D':Operation.TYPES.ACTION,
+        'Count':Operation.TYPES.ACTION,
+        'Min':Operation.TYPES.ACTION,
+        'Max':Operation.TYPES.ACTION,
+        'Mean':Operation.TYPES.ACTION,
+        'Sum':Operation.TYPES.ACTION,
+        'Fill':Operation.TYPES.ACTION,
         }
         
-        return operations_dict.get(name)
+        op_type = operations_dict.get(name)
+
+        if not op_type:
```

Comment:
Remember to add here a case for `Take` and `Snapshot`, the message in that case should be different, since they are not invalid, just not supported for now. You can have a separate dictionary for the non-supported operations and throw you error depending on that.

Reply:
I think it's `Take` and `Foreach`.
`Snapshot` should be available in python afaik.

Reply:
@etejedor , @bluehood , I've created a new list instead 👍

Reply:
@shravan97 I don't see `Snapshot` and `Foreach` anywhere..?
They can both go together with `Range` and `Take` (currently not supported).

Reply:
Yep sure. Will add this :+1: